src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImplTest.kt

```diff
@@ -206,4 +208,31 @@ class InterventionFilterRepositoryImplTest @Autowired constructor(
       assertThat(it.dynamicFrameworkContract.maximumAge).isEqualTo(29)
     }
   }
+
+  @Test
+  fun `get interventions with show-future-interventions disabled `() {
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(pccRegion = pccRegionFactory.create(), referralStartDate = LocalDate.now().plusDays(10)))
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(pccRegion = pccRegionFactory.create(), referralStartDate = LocalDate.now()))
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(pccRegion = pccRegionFactory.create()), title = "test Title")
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(npsRegion = npsRegionFactory.create()))
+    val found = interventionFilterRepositoryImpl.findByCriteria(listOf(), null, null, null, null)
+
+    assertThat(found.size).isEqualTo(3)
+    found.forEach {
+      assertThat(it.dynamicFrameworkContract.referralStartDate).isBeforeOrEqualTo(LocalDate.now())
+    }
+  }
+
+  @Test
+  fun `get interventions with show-future-interventions enabled `() {
+    ReflectionTestUtils.setField(interventionFilterRepositoryImpl, "showFutureInterventions", true)
+
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(pccRegion = pccRegionFactory.create(), referralStartDate = LocalDate.now().plusDays(10)))
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(pccRegion = pccRegionFactory.create(), referralStartDate = LocalDate.now()))
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(pccRegion = pccRegionFactory.create()), title = "test Title")
+    interventionFactory.create(contract = dynamicFrameworkContractFactory.create(npsRegion = npsRegionFactory.create()))
+    val found = interventionFilterRepositoryImpl.findByCriteria(listOf(), null, null, null, null)
+
+    assertThat(found.size).isEqualTo(4)
```

Comment:
_cognitive load_

Would this test break, I would wonder how to fix it, e.g. why it's not `4`. Could it be explicit that a "future" contract is not selected? E.g. `found` must include `futureContract` or similar?